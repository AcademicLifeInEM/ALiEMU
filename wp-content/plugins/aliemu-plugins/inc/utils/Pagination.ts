
/**
 * Takes an Object and returns an array containing the object params that should
 *   be visible on the current page.
 *
 * @param  {Object} rows Any variable that has the Object prototype. The data
 *   for the 'rows' of interest should be contained within child objects of the
 *   'row' object
 * @param  {number} visibleRows Number of rows that should be visible.
 * @param  {number} currentPage Current selected page.
 * @return {string[]} The object keys of the the rows that should be visible.
 */
export default function paginate(rows: Array<any>, visibleRows: number, currentPage: number) {
    return rows.filter((uid, i: number) =>
        (visibleRows * currentPage) <= i && i < (visibleRows * currentPage + visibleRows)
    );
}
