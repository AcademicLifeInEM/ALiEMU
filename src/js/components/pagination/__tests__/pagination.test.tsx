import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('lodash', () => ({
    uniqueId: jest.fn(id => `${id}TEST`),
}));

import Pagination from '../index';

describe('<Pagination />', () => {
    beforeAll(() => {
        Date.now = jest.fn(() => 0);
    });
    it('should render correctly', () => {
        const tree = renderer.create(
            <Pagination total={5} value={3} onChange={jest.fn()} />,
        );
        expect(tree).toMatchSnapshot();
    });
});
