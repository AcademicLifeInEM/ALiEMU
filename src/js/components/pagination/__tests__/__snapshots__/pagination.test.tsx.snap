// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render correctly 1`] = `
<nav
  aria-labelledby="pagination-TEST"
  className="nav alignRight"
  role="navigation"
>
  <div
    className="pager"
  >
    <span
      aria-live="polite"
      role="status"
    >
      <h2
        className="clipped"
        id="pagination-TEST"
      >
        Table Pagination - Page 3
      </h2>
    </span>
    <button
      aria-label="previous page"
      className="buttonPrev"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "fontSize": 15,
          "height": 30,
          "minWidth": 30,
        }
      }
    >
      <i
        className="material-icons"
        style={
          Object {
            "color": undefined,
            "fontSize": 16,
          }
        }
      >
        chevron_left
      </i>
    </button>
    <div
      className="pageList"
      role="list"
    >
      <button
        aria-label="page 1"
        className="button"
        onClick={[Function]}
        style={
          Object {
            "fontSize": 15,
            "height": 30,
            "minWidth": 30,
          }
        }
      >
        1
      </button>
      <button
        aria-label="page 2"
        className="button"
        onClick={[Function]}
        style={
          Object {
            "fontSize": 15,
            "height": 30,
            "minWidth": 30,
          }
        }
      >
        2
      </button>
      <button
        aria-current="page"
        aria-label="page 3"
        className="button"
        onClick={[Function]}
        style={
          Object {
            "fontSize": 15,
            "height": 30,
            "minWidth": 30,
          }
        }
      >
        3
      </button>
      <button
        aria-label="page 4"
        className="button"
        onClick={[Function]}
        style={
          Object {
            "fontSize": 15,
            "height": 30,
            "minWidth": 30,
          }
        }
      >
        4
      </button>
      <button
        aria-label="page 5"
        className="button"
        onClick={[Function]}
        style={
          Object {
            "fontSize": 15,
            "height": 30,
            "minWidth": 30,
          }
        }
      >
        5
      </button>
    </div>
    <button
      aria-label="next page"
      className="buttonNext"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "fontSize": 15,
          "height": 30,
          "minWidth": 30,
        }
      }
    >
      <i
        className="material-icons"
        style={
          Object {
            "color": undefined,
            "fontSize": 16,
          }
        }
      >
        chevron_right
      </i>
    </button>
  </div>
</nav>
`;
